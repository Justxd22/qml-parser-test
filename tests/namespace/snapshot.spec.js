import fs from "fs";
import { parse } from "../../src/index.js";
import { fileURLToPath } from "url";
import { dirname } from "path";

// Get __dirname equivalent in ES Modules
const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

it("renders correctly", () => {
  const ast = parse(fs.readFileSync(`${__dirname}/entry.qml`).toString());

  expect(ast).toMatchSnapshot();
});
