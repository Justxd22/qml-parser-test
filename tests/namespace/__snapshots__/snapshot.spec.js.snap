// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
{
  "children": [
    {
      "as": "QQ",
      "identifier": "QtQuick",
      "kind": "Import",
      "loc": {
        "end": {
          "column": 24,
          "line": 1,
          "offset": 24,
        },
        "start": {
          "column": 0,
          "line": 1,
          "offset": 0,
        },
      },
      "version": "2.0",
    },
    {
      "children": [
        {
          "identifier": "test1",
          "kind": "Property",
          "loc": {
            "end": {
              "column": 31,
              "line": 5,
              "offset": 73,
            },
            "start": {
              "column": 2,
              "line": 5,
              "offset": 44,
            },
          },
          "type": "QQ.Item",
          "value": {
            "kind": "JavascriptValue",
            "loc": {
              "end": {
                "column": 31,
                "line": 5,
                "offset": 73,
              },
              "start": {
                "column": 27,
                "line": 5,
                "offset": 69,
              },
            },
            "object": false,
            "value": "null",
          },
        },
        {
          "identifier": "test2",
          "kind": "Property",
          "loc": {
            "end": {
              "column": 31,
              "line": 7,
              "offset": 107,
            },
            "start": {
              "column": 2,
              "line": 7,
              "offset": 78,
            },
          },
          "type": "QQ.Text",
          "typeModifier": "list",
        },
        {
          "asBrackets": true,
          "identifier": "test3",
          "kind": "Signal",
          "loc": {
            "end": {
              "column": 29,
              "line": 9,
              "offset": 138,
            },
            "start": {
              "column": 2,
              "line": 9,
              "offset": 111,
            },
          },
          "parameters": [
            {
              "identifier": "arg",
              "type": "QQ.Image",
            },
          ],
        },
      ],
      "identifier": "QQ.Rectangle",
      "kind": "ObjectDefinition",
      "loc": {
        "end": {
          "column": 1,
          "line": 11,
          "offset": 142,
        },
        "start": {
          "column": 0,
          "line": 3,
          "offset": 26,
        },
      },
    },
  ],
  "comments": [],
  "kind": "Program",
  "loc": {
    "end": {
      "column": 0,
      "line": 12,
      "offset": 143,
    },
    "start": {
      "column": 0,
      "line": 1,
      "offset": 0,
    },
  },
}
`;
